File: bindings/lua/Polycode/SceneParticle.lua
class "SceneParticle"


function SceneParticle:__getvar(name)
	if name == "lifetime" then
		return Polycode.SceneParticle_get_lifetime(self.__ptr)
	elseif name == "position" then
		local retVal = Polycode.SceneParticle_get_position(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "velocity" then
		local retVal = Polycode.SceneParticle_get_velocity(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "perlinPos" then
		local retVal = Polycode.SceneParticle_get_perlinPos(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "rotation" then
		local retVal = Polycode.SceneParticle_get_rotation(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Vector3"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "brightnessDeviation" then
		return Polycode.SceneParticle_get_brightnessDeviation(self.__ptr)
	elseif name == "scale" then
		return Polycode.SceneParticle_get_scale(self.__ptr)
	elseif name == "color" then
		local retVal = Polycode.SceneParticle_get_color(self.__ptr)
		if retVal == nil then return nil end
		local __c = _G["Color"]("__skip_ptr__")
		__c.__ptr = retVal
		return __c
	elseif name == "varianceIndex" then
		return Polycode.SceneParticle_get_varianceIndex(self.__ptr)
	end
end

function SceneParticle:__setvar(name,value)
	if name == "lifetime" then
		Polycode.SceneParticle_set_lifetime(self.__ptr, value)
		return true
	elseif name == "position" then
		Polycode.SceneParticle_set_position(self.__ptr, value.__ptr)
		return true
	elseif name == "velocity" then
		Polycode.SceneParticle_set_velocity(self.__ptr, value.__ptr)
		return true
	elseif name == "perlinPos" then
		Polycode.SceneParticle_set_perlinPos(self.__ptr, value.__ptr)
		return true
	elseif name == "rotation" then
		Polycode.SceneParticle_set_rotation(self.__ptr, value.__ptr)
		return true
	elseif name == "brightnessDeviation" then
		Polycode.SceneParticle_set_brightnessDeviation(self.__ptr, value)
		return true
	elseif name == "scale" then
		Polycode.SceneParticle_set_scale(self.__ptr, value)
		return true
	elseif name == "color" then
		Polycode.SceneParticle_set_color(self.__ptr, value.__ptr)
		return true
	elseif name == "varianceIndex" then
		Polycode.SceneParticle_set_varianceIndex(self.__ptr, value)
		return true
	end
	return false
end
function SceneParticle:__delete()
	if self then Polycode.delete_SceneParticle(self.__ptr) end
end

File: bindings/javascript/Polycode/SceneParticle.js
function SceneParticle() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.SceneParticle()
	}
	Object.defineProperties(this, {
		'lifetime': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_lifetime, set: SceneParticle.prototype.__set_lifetime},
		'position': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_position, set: SceneParticle.prototype.__set_position},
		'velocity': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_velocity, set: SceneParticle.prototype.__set_velocity},
		'perlinPos': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_perlinPos, set: SceneParticle.prototype.__set_perlinPos},
		'rotation': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_rotation, set: SceneParticle.prototype.__set_rotation},
		'brightnessDeviation': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_brightnessDeviation, set: SceneParticle.prototype.__set_brightnessDeviation},
		'scale': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_scale, set: SceneParticle.prototype.__set_scale},
		'color': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_color, set: SceneParticle.prototype.__set_color},
		'varianceIndex': { enumerable: true, configurable: true, get: SceneParticle.prototype.__get_varianceIndex, set: SceneParticle.prototype.__set_varianceIndex}
	})
}


SceneParticle.prototype.__get_lifetime = function() {
	return Polycode.SceneParticle__get_lifetime(this.__ptr)
}

SceneParticle.prototype.__set_lifetime = function(val) {
	Polycode.SceneParticle__set_lifetime(this.__ptr, val)
}

SceneParticle.prototype.__get_position = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.SceneParticle__get_position(this.__ptr)
	return retVal
}

SceneParticle.prototype.__set_position = function(val) {
	Polycode.SceneParticle__set_position(this.__ptr, val.__ptr)
}

SceneParticle.prototype.__get_velocity = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.SceneParticle__get_velocity(this.__ptr)
	return retVal
}

SceneParticle.prototype.__set_velocity = function(val) {
	Polycode.SceneParticle__set_velocity(this.__ptr, val.__ptr)
}

SceneParticle.prototype.__get_perlinPos = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.SceneParticle__get_perlinPos(this.__ptr)
	return retVal
}

SceneParticle.prototype.__set_perlinPos = function(val) {
	Polycode.SceneParticle__set_perlinPos(this.__ptr, val.__ptr)
}

SceneParticle.prototype.__get_rotation = function() {
	var retVal = new Vector3("__skip_ptr__")
	retVal.__ptr = 	Polycode.SceneParticle__get_rotation(this.__ptr)
	return retVal
}

SceneParticle.prototype.__set_rotation = function(val) {
	Polycode.SceneParticle__set_rotation(this.__ptr, val.__ptr)
}

SceneParticle.prototype.__get_brightnessDeviation = function() {
	return Polycode.SceneParticle__get_brightnessDeviation(this.__ptr)
}

SceneParticle.prototype.__set_brightnessDeviation = function(val) {
	Polycode.SceneParticle__set_brightnessDeviation(this.__ptr, val)
}

SceneParticle.prototype.__get_scale = function() {
	return Polycode.SceneParticle__get_scale(this.__ptr)
}

SceneParticle.prototype.__set_scale = function(val) {
	Polycode.SceneParticle__set_scale(this.__ptr, val)
}

SceneParticle.prototype.__get_color = function() {
	var retVal = new Color("__skip_ptr__")
	retVal.__ptr = 	Polycode.SceneParticle__get_color(this.__ptr)
	return retVal
}

SceneParticle.prototype.__set_color = function(val) {
	Polycode.SceneParticle__set_color(this.__ptr, val.__ptr)
}

SceneParticle.prototype.__get_varianceIndex = function() {
	return Polycode.SceneParticle__get_varianceIndex(this.__ptr)
}

SceneParticle.prototype.__set_varianceIndex = function(val) {
	Polycode.SceneParticle__set_varianceIndex(this.__ptr, val)
}

Duktape.fin(SceneParticle.prototype, function (x) {
	if (x === SceneParticle.prototype) {
		return;
	}
	Polycode.SceneParticle__delete(x.__ptr)
})
